### ISSUE TITLE
Lots of lag after extended play

### ISSUE THREAD

**Issue by pkmnfrk** [pkmnfrk/equivalentintegrations#5]
Unsure if this is EI or another mod, but after playing for a while (3-4 hours), the game starts hanging for about 3 seconds every 10 seconds. I think it's severe GC, which might be due to the fact that we throw out the cached knowledge array any time it changes, which can potentially happen at least a few times every tick.

**Comment by pkmnfrk** [comment]
I've completely redone how the inventory is calculated, and centralized it per player (since it's going to be the same for every chamber). This should greatly reduce the memory pressure, if indeed this is what was causing it. But, I need to do a few hours of testing to confirm

**Comment by pkmnfrk** [comment]
I left the game running for about 5 hours, and when I came back, it was basically unresponsive. Investigating the logs, the following is true:
* The game started at about 15:40
* I stopped actively playing around 17:00
* The first "Can't keep up!" message occurs at 18:15, and reoccurs every ~15 seconds thereafter
* As time progresses, the "running behind" time gets larger, starting around 2-3 seconds, eventually getting up to 40-50 seconds around 20:50
* I'm not sure if this was caused by my return, but as soon as I sat down at 21:00 and saw the state of the game, I tried to pause. This caused the game to lock up entirely and crash with an OutOfMemoryException

This is definitely a memory leak of some kind, but I'm not sure where it's coming from. I'm having a hard time believing that it's this mod, since I basically removed all dynamic allocations other than ItemStacks. What could it be?
